File: src/main/java/aurocosh/divinefavor/common/block/ethereal/BlockEtherealFlash.java
package aurocosh.divinefavor.common.block.ethereal;

import aurocosh.divinefavor.DivineFavor;
import net.minecraft.block.material.Material;
import net.minecraft.block.state.IBlockState;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;

import java.util.Random;

public class BlockEtherealFlash extends BlockEtherealLight {
    private final int despawnDelay;

    public BlockEtherealFlash(String name, int lightLevel,  int despawnDelay) {
        super(name, Material.ROCK, lightLevel);
        this.despawnDelay = despawnDelay;
        setCreativeTab(DivineFavor.TAB_MAIN);
    }

    @Override
    public void onBlockAdded(World worldIn, BlockPos pos, IBlockState state) {
        worldIn.scheduleUpdate(pos, this, despawnDelay);
    }

    @Override
    public void updateTick(World worldIn, BlockPos pos, IBlockState state, Random rand) {
        worldIn.setBlockToAir(pos);
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/block/ethereal/BlockEtherealFlash.kt
package aurocosh.divinefavor.common.block.ethereal

import aurocosh.divinefavor.DivineFavor
import net.minecraft.block.material.Material
import net.minecraft.block.state.IBlockState
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World
import java.util.*

class BlockEtherealFlash(name: String, lightLevel: Int, private val despawnDelay: Int) : BlockEtherealLight(name, Material.ROCK, lightLevel) {

    init {
        creativeTab = DivineFavor.TAB_MAIN
    }

    override fun onBlockAdded(worldIn: World, pos: BlockPos, state: IBlockState) {
        worldIn.scheduleUpdate(pos, this, despawnDelay)
    }

    override fun updateTick(worldIn: World, pos: BlockPos, state: IBlockState, rand: Random) {
        worldIn.setBlockToAir(pos)
    }
}